dot/sync/bootstrap_syncer_integration_test.go

```diff
@@ -28,9 +28,10 @@ func newTestBootstrapSyncer(t *testing.T) *bootstrapSyncer {
 		trie.EmptyHash, 200, types.NewDigest())
 	require.NoError(t, err)
 
-	bs := new(syncmocks.BlockState)
-	bs.On("BestBlockHeader").Return(header, nil)
-	bs.On("GetHighestFinalisedHeader").Return(finHeader, nil)
+	ctrl := gomock.NewController(t)
+	bs := NewMockBlockState(ctrl)
+	bs.EXPECT().BestBlockHeader().Return(header, nil).AnyTimes()
```

Comment:
Could you set a specific amount of time the function should be called, this makes the test more strict and I think that `.AnyTimes()` allows a function to not be called

Reply:
That's related to #2659 

We previously didn't even assert the behavior of the mockery mock, so it could be called 0, 1, 2.. times and we would not even know (ptsd).

Now yes fixing to a certain number of times would be a nice addition indeed, especially when it comes to mocks defined in a test (not a (horrible) mock test helper).